docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1680,23 +1572,165 @@ what does traditional mean here?   Round robin?
 TITLE: Threads
 
 [REQUIREMENT]
-UID: ZEP-120
+UID: ZEP-23
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Threads
-TITLE: Thread Support
+TITLE: Creating threads
 STATEMENT: >>>
-Zephyr shall support multiple threads of execution.
+Zephyr shall provide an interface to create (start) a thread.
 <<<
-USER_STORY: >>>
-As a Zephyr OS user, I want to be able to execute multiple threads which are managed by priorities or other mechanisms.
+DISCUSSION_DATE: >>>
+pf-ok
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-107
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Setting thread priority
+STATEMENT: >>>
+Zephyr shall provide an interface to set a thread's priority.
+<<<
+DISCUSSION_DATE: >>>
+pf-ok
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-122
+
+[REQUIREMENT]
+UID: ZEP-108
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Suspending a thread
+STATEMENT: >>>
+Zephyr shall provide an interface to suspend a thread.
+<<<
+DISCUSSION_DATE: >>>
+pf-ok
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-109
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Resuming a suspended thread
+STATEMENT: >>>
+Zephyr shall provide an interface to resume a suspended thread.
+<<<
+DISCUSSION_DATE: >>>
+pf-ok
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-110
+STATUS: Draft
+TYPE: Functional
+TITLE: Resuming a suspended thread after a timeout
+STATEMENT: >>>
+Zephyr shall provide an interface to resume a suspended thread after a timeout.
+<<<
+DISCUSSION_DATE: >>>
+pf-ok
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-111
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Deleting a thread
+STATEMENT: >>>
+Zephyr shall provide an interface to delete (end) a thread.
 <<<
 DISCUSSION_DATE: >>>
-pf-pq
+pf-ok
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-124
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Thread states
+STATEMENT: >>>
+Threads shall have different states to fulfill the Lifecycle of a thread
+<<<
+USER_STORY: >>>
+As a Zephyr OS user, I want to know in what state a specific thread is.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-98
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Thread stack objects
+STATEMENT: >>>
+Every Thread shall have it's own stack.
+<<<
+USER_STORY: >>>
+As a Zephyr OS user I want to be able to configure the stack size of a thread. And every thread shall have it's own dedicated stack.
```

Comment:
**nitpick:** Whenever I see AND in some RQT it's probably two of them.

Reply:
True but in this case it is in the user story why this RQT exists and not in the statement itself.